lib/game.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+require_relative '../dependencies'
+
+class Game
+  include Validation
+  SECRET_CODE_LENGTH = 4
+  RANGE_START = 1
+  RANGE_END = 6
+  NOT_YET = '-'
+  GOT_IT = '+'
+  DIFFICULTY_LEVEL = {
+    easy: { attempts: 15, hints: 3 },
+    medium: { attempts: 10, hints: 2 },
+    hell: { attempts: 5, hints: 1 }
+  }.freeze
+
+  attr_accessor :attempts_total, :attempts, :difficulty, :hints_total, :hints, :name, :win, :secret
+
+  def initialize(name:, difficulty:)
+    @name = name
+    @difficulty = difficulty
+    @attempts = DIFFICULTY_LEVEL[difficulty][:attempts]
+    @hints = DIFFICULTY_LEVEL[difficulty][:hints]
+    @win = false
+    @secret = make_number
+    @unused_hints = @secret.chars
+  end
+
+  def make_number(numbers = RANGE_END)
+    (1..SECRET_CODE_LENGTH).map { rand(RANGE_START..numbers) }.join
+  end
+
+  def check(number)
+    @attempts -= 1
+    result = check_numbers(@secret.chars, number.chars)
+    win?(result)
+  end
+
+  def win?(result)
```

Comment:
methods with ? should only return a boolean, they should not contain any logic
- please remove `@result = result` (@result is not used at all)
- assign @win variable in `check` method